airflow/operators/dagrun_operator.py

```diff
@@ -126,3 +144,22 @@ def execute(self, context: Dict):
                 dag.clear(start_date=self.execution_date, end_date=self.execution_date)
             else:
                 raise e
+
+        if self.wait_for_completion:
+            # wait for dag to complete
+            while True:
+                self.log.info(
+                    'Waiting for %s on %s to become allowed state %s ...',
+                    self.trigger_dag_id,
+                    dag_run.execution_date,
+                    self.allowed_states,
+                )
+                dag_run.refresh_from_db()
+                state = dag_run.state
+                if state in self.failed_states:
+                    raise AirflowException(f"{self.trigger_dag_id} failed with failed states {state}")
+                elif state in self.allowed_states:
+                    self.log.info("%s finished with allowed state %s", self.trigger_dag_id, state)
+                    return
+
+                time.sleep(self.poke_interval)
```

Comment:
Let's move the sleep to the beginning of the loop. It's a slight improvement but may stucture the code better

Reply:
ok i can